backend/cmm_helpers.ml

```diff
@@ -772,8 +772,18 @@ let complex_im c dbg =
 
 let return_unit dbg c = Csequence (c, Cconst_int (1, dbg))
 
-let field_address ptr n dbg =
-  if n = 0 then ptr else Cop (Cadda, [ptr; Cconst_int (n * size_addr, dbg)], dbg)
+let field_address ?(memory_chunk = Word_val) ptr n dbg =
+  let field_size_in_bytes =
```

Comment:
this let-binding could be moved down into the `else` (i.e. `n <> 0`) branch of the conditional that is right after.